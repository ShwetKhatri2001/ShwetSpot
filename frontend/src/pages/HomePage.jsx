import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Col, Row } from 'react-bootstrap';
import Product from '../components/Product';
import Message from '../components/Message';
import Loader from '../components/Loader';
import { listProducts } from '../redux/actions/productActions';

const HomePage = () => {
  const dispatch = useDispatch();

  const productList = useSelector(state => state.productList);
  const { loading, error, products } = productList;

  const userLogin = useSelector(state => state.userLogin);
  const { userInfo } = userLogin;

  useEffect(() => {
    dispatch(listProducts());
  }, [dispatch, userInfo]);

  return (
    <>
      <h1 id="recently-added">Recently Added</h1>
      {loading ? (
        <Loader />
      ) : error ? (
        <Message variant="danger">{error}</Message>
      ) : (
        <>
          <Row>
            {products.map(product => (
              <Col
                key={product._id}
                sm={12}
                md={6}
                lg={4}
                xl={3}
                className="mb-2"
              >
                <Product product={product} />
              </Col>
            ))}
          </Row>
        </>
      )}
    </>
  );
};

export default HomePage;
