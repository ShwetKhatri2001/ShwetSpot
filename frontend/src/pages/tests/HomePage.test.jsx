import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import HomePage from '../HomePage';
import store from '../../redux/store';

describe('HomePage Component', () => {
  const MockHomePage = (
    <MemoryRouter>
      <Provider store={store}>
        <HomePage />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockHomePage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
