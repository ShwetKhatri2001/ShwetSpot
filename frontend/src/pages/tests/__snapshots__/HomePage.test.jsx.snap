// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should match the snapshot 1`] = `
Array [
  <h1
    id="recently-added"
  >
    Recently Added
  </h1>,
  <div
    className="row"
  />,
]
`;
